vm/string.go

```diff
@@ -0,0 +1,11 @@
+package vm
+
+import "C"
+import "unsafe"
+
+// cstring creates a null-terminated C string from the given byte slice.
+// the caller is responsible for freeing the underlying memory
+func cstring(data []byte) *C.char {
```

Comment:
By the way, are UTF-8 strings important for us here? 🤔
cc @omerfirmak

Reply:
I believe so. See [here](https://eqlabs.github.io/pathfinder/blog/2023-06-17_mainnet_incident.html#root-cause-of-the-issue)

Reply:
I rechecked `C.CString` implementation:
```
const char *_cgoPREFIX_Cfunc_CString(struct __go_string s) {
	char *p = malloc(s.__length+1);
	if(p == NULL)
		runtime_throw("runtime: C malloc failed");
	memmove(p, s.__data, s.__length);
	p[s.__length] = 0;
	return p;
}
```
which seems fine for `unsafe.String(...)` trick.

Regarding UTF-8: in Rust if you create string with invalid UTF-8 symbol it will panic, but since json valid can contain only valid UTF-8 chars we're fine.